### ISSUE TITLE
Make sphinx autosummary and autoclass consistent

### ISSUE THREAD

**Issue by jrbourbeau** [dask/distributed#3693]
In our API docs we currently use 

```
.. autosummary::
   Client
   Client.call_stack
   Client.cancel
    ...
```

To generate a table of `Client` methods at the [top of the page](https://distributed.dask.org/en/latest/api.html#api). Later on we use 

```
.. autoclass:: Client
   :members:
```

to display the docstrings for all the public methods on `Client` (here an example for [`Client.gather`](https://distributed.dask.org/en/latest/api.html#distributed.Client.gather)). However, sometimes these can get [out of sync](https://github.com/dask/distributed/issues/3691#issuecomment-611741612) if there is a method that appears in `.. autoclass:: Client`, but isn't added to the `.. autosummary::` section at the top of the page. It would be great to see if there's a way to automatically ensure these two sections are consistent with one another (i.e. that the same methods are in both the `autosummary` and `autoclass` sections).

**Comment by mrocklin** [comment]
+1 on finding the right way in sphinx to automatically generate these
lists.  They're a common papercut when people add new methods/functions.

**Comment by caseyclements** [comment]
Hi. I picked this up. My firm is having a doing an open-source evangelism week..

PR's here: https://github.com/dask/distributed/pull/4367

**Comment by gforsyth** [comment]
Closed via #4367
